health-services/plan-service/src/main/java/digit/service/validator/PlanEmployeeAssignmentValidator.java

```diff
@@ -205,11 +195,70 @@ private void validateCampaignDetails(String campaignId, String tenantId, PlanEmp
         // Validate if campaign id exists against project factory
         validateCampaignId(campaignResponse);
 
-        // Validate the provided jurisdiction for employee
-        validateEmployeeJurisdiction(campaignResponse.getCampaignDetails().get(0), planEmployeeAssignment);
+        // Validate the provided jurisdiction for the plan employee assignment
+        validateEmployeeAssignmentJurisdiction(campaignResponse.getCampaignDetails().get(0), planEmployeeAssignment);
+
+        // Validates the jurisdiction assigned to Root role employee against MDMS
+        validateRootEmployeeJurisdiction(planEmployeeAssignment, mdmsData, campaignResponse.getCampaignDetails().get(0));
 
-        // Validates highest root jurisdiction for National roles against MDMS
-        validateNationalRole(planEmployeeAssignment, mdmsData, campaignResponse.getCampaignDetails().get(0));
+        // Validates the jurisdiction assigned to a non-Root employee against MDMS
+        validateEmployeeJurisdiction(planEmployeeAssignment, mdmsData, campaignResponse.getCampaignDetails().get(0));
+    }
+
+    /**
+     * This is a helper method to get the lowest or highest hierarchy for microplan from MDMS
+     *
+     * @param mdmsData        the mdms data
+     * @param lowestHierarchy if true, returns the lowest hierarchy for microplan
+     * @return returns the lowest or highest hierarchy for microplan
+     */
+    private String getMicroplanHierarchy(Object mdmsData, boolean lowestHierarchy) {
+
+        String jsonPathForMicroplanHierarchy = JSON_ROOT_PATH + MDMS_ADMIN_CONSOLE_MODULE_NAME + DOT_SEPARATOR + MDMS_MASTER_HIERARCHY_CONFIG + HIERARCHY_CONFIG_FOR_MICROPLAN;
+
+        List<Map<String, String>> hierarchyForMicroplan;
```

Comment:
see if the jsonpath can just return the object instead of an array of object since you are already filtering by heirarchy = MICROPLAN